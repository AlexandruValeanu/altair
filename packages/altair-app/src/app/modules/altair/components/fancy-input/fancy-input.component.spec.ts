import { expect, it, beforeEach, describe } from '@jest/globals';
import { ComponentFixture, TestBed, tick, fakeAsync, waitForAsync } from '@angular/core/testing';

import { FancyInputComponent } from './fancy-input.component';
import { FormsModule } from '@angular/forms';
import { SharedModule } from '../../modules/shared/shared.module';
import { MockModule } from 'ng-mocks';
import { Store } from '@ngrx/store';
import { mock, mockStoreFactory, mount, NgxTestWrapper } from '../../../../../testing';
import { FancyInputMarkerComponent } from '../fancy-input-marker/fancy-input-marker.component';
import { EnvironmentService } from '../../services';
import { NO_ERRORS_SCHEMA } from '@angular/core';

describe('FancyInputComponent', () => {
  let wrapper: NgxTestWrapper<FancyInputComponent>;

  beforeEach(async() => {
    wrapper = await mount({
      component: FancyInputComponent,
      imports: [
        MockModule(SharedModule),
      ],
      providers: [
        {
          provide: Store,
          useValue: mockStoreFactory({
            settings: {}
          }),
        },
        {
          provide: EnvironmentService,
          useValue: mock({
            getActiveEnvironment() { return {} }
          }),
        },
      ],
      schemas: [ NO_ERRORS_SCHEMA ],
    });
  })

  it('should create', () => {
    expect(wrapper.componentInstance).toBeTruthy();
  });

  it('should set component value when input value changes', async() => {
    const input = wrapper.find('input');
    input.setValue('some text');
    await input.nextTick();

    expect(wrapper.element).toMatchSnapshot();
    // TODO: Figure this out
    // expect(wrapper.componentInstance.value).toBe('some text');
  });

  it('should render the highlights correctly based on the matched variables in text', async() => {
    const input = wrapper.find('input');
    input.setValue('some text {{variable1}} then {{variable2}} is next');
    await input.nextTick();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render mark element containing the matched variables in text', async() => {
    const input = wrapper.find('input');
    input.setValue('some text {{variable1}} then {{variable2}} is next');
    await input.nextTick();

    // TODO: Figure this out
    // const marks = wrapper.findAll('mark');

    // expect(marks.length).toBe(2);
    // expect(marks[0].text()).toBe('{{variable1}}');
    // expect(marks[1].text()).toBe('{{variable2}}');
  });
});
