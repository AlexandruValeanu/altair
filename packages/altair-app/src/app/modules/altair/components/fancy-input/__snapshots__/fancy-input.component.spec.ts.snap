// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FancyInputComponent should render the highlights correctly based on the matched variables in text 1`] = `
<app-fancy-input>
  <div
    class="fancy-input-container"
  >
    <input
      class="fancy-input-content fancy-input-element"
      placeholder=""
      type="text"
    />
    <div
      class="fancy-input-backdrop"
    >
      <div
        class="fancy-input-content fancy-input-highlights"
      >
        <span>
          some text 
        </span>
        
        
        
        
        <app-fancy-input-marker />
        
        
        <span>
           then 
        </span>
        
        
        
        
        <app-fancy-input-marker />
        
        
        <span>
           is next
        </span>
        
        
        
        
      </div>
    </div>
  </div>
</app-fancy-input>
`;

exports[`FancyInputComponent should set component value when input value changes 1`] = `
<app-fancy-input>
  <div
    class="fancy-input-container"
  >
    <input
      class="fancy-input-content fancy-input-element"
      placeholder=""
      type="text"
    />
    <div
      class="fancy-input-backdrop"
    >
      <div
        class="fancy-input-content fancy-input-highlights"
      >
        <span>
          some text
        </span>
        
        
        
        
      </div>
    </div>
  </div>
</app-fancy-input>
`;
